lib/_tls_wrap.js

```diff
@@ -409,20 +410,23 @@ TLSSocket.prototype._wrapHandle = function(wrap) {
   res._secureContext = context;
   res.reading = handle.reading;
   this[kRes] = res;
-  Object.defineProperty(handle, 'reading', {
-    get: () => {
-      return this[kRes].reading;
-    },
-    set: (value) => {
-      this[kRes].reading = value;
-    }
-  });
+  defineHandleReading(this, handle);
 
   this.on('close', onSocketCloseDestroySSL);
 
   return res;
 };
 
+function defineHandleReading(socket, handle) {
```

Comment:
Would you mind adding a comment that links to https://github.com/nodejs/node/commit/f7620fb96d339f704932f9bb9a0dceb9952df2d4, explaining why it is being done in this way?

Reply:
Updated, I think the commit explains it well (and it shows all the relevant parts) but let me know if you prefer a full outline of it within the code.